### PR TITLE
Kien-tasks1

### PR DESCRIPTION
**Pull request by hayate0304**
RESULTS.PY

- Fix normed deprecated by replacing normed kwarg with density kwarg.
- Fix Hist normalize=False should return bin counts.
- Add .summary() method: able to display mean(), sd() and plot(), not able to display any tables yet (will come back to implement this later). 
- Add .min(), .max(): just realized that the old pull request has the same with mine.

PLOT.PY
_ Change tile plot so intensity scale starts at 0: I add vmin = 0 at ax.matshow. It works for me but not sure if this is what you mean.

DISTRIBUTIONS.PY

- Normal: change default parameter from `var` to `sd`
- Beta: add xmin, xmax parameters

### CODE REVIEW

File: symbulate/distributions.py [hunk 1]
```diff
@@ -375,25 +375,17 @@ class Normal(Distribution):
...
-        if sd is None:
-            if var > 0:
-                self.scale = np.sqrt(var)
-            elif var == 0: 
-                raise NotImplementedError
-                #TODO
-            else:
-                raise Exception("var cannot be less than 0")
-        
+        if var is not None:
```
Comment (dlsun) [inline symbulate/distributions.py]: I think this is wrong. I should be able to do `Normal(mean=0, var=1)`, but I can't. You need to allow the user to specify `sd=` or `var=`, their choice.

File: symbulate/distributions.py [hunk 2]
```diff
@@ -533,16 +528,16 @@ def __init__(self, a, b, scale=None):
         self.xlim = (0, 1) # Beta distributions are not defined for x < 0 and x > 1
 
     def draw(self):
-        """A function that takes no arguments and 
+        """A function that takes no arguments and
             returns a single draw from the Beta distribution."""
 
-        return np.random.beta(self.a, self.b)
+        return (self.xmin + (self.xmax - self.xmin)*np.random.beta(self.a, self.b))
```
Comment (dlsun) [inline symbulate/distributions.py]: Remove outer parentheses, make sure you have spaces surrounding all operations, like `*`.

File: symbulate/results.py [hunk 3]
```diff
@@ -473,9 +499,36 @@ def sd(self):
...
+
+    def max(self):
+        if all(is_scalar(x) for x in self):
+            return np.array(self).max()
+        elif get_dimension(self) > 0:
+            return tuple(np.array(self).max(0))
+        else:
+            raise Exception("I don't know how to take the max of these values.")
+
+    def summary(self):
```
Comment (dlsun) [inline symbulate/results.py]: Dr. Ross suggested that this be a table. In its current form, it doesn't seem very useful.

File: symbulate/distributions.py (hunk 1 above)
Comment (hayate0304) [reply symbulate/distributions.py]: Thanks professor. I fixed and tested again. 
image

File: symbulate/results.py (hunk 3 above)
Comment (hayate0304) [reply symbulate/results.py]: I removed it.

**Comment by hayate0304** [comment]
But then can something like this happen. Since `var is not None` and sd is also specified. I'm not sure how can I raise Exception when both are provided. 
image

**Comment by hayate0304** [comment]
Ahhh, I got it, I added 2 lines and tested and it works

image

**Comment by kevindavisross** [comment]
A notebook with my review of these changes is attached.  Two minor revisions are required

- Normal var vs sd is working properly, but the comment in the docstring has it backwards.  It should be "sd is ignored if var is provided"; see `?BivariateNormal`
- For Beta, add `a=1, b=1` as default parameters (so default Beta = Uniform)

To be clear, the changes that this pull requests implements are
- Tile: fix plot so color axis scale starts at 0
- Hist: fix "depricated" warnings
- Hist: normalize=False option returns counts, for both 1d and 2d hist
- Beta: allow support on (xmin, xmax) and not just (0,1)
- Normal: change default from var to sd, e.g. Normal(0,4) has sd=4.

If there are other changes that I did not mention, please let me know.

A few notes for later
- marginal option with 2dhist and normalize=False needs work
- Beta pdf plot has (0,1) as support even for other (xmin, xmax).  Needs to be fixed; see Cauchy (which seems to work correctly).
[Test - Kien-tasks - histograms - tile - Normal var - Beta.pdf](https://github.com/dlsun/symbulate/files/2294750/Test.-.Kien-tasks.-.histograms.-.tile.-.Normal.var.-.Beta.pdf)

**Comment by kevindavisross** [comment]
This pull request looks good to me.  One question: for Beta (xmin, xmax) together define the interval, currently without requiring xmin < xmax.  For example, xmin=10, xmax=1 will create Beta with support on (1, 10) without error.  This isn't too bad, but it is inconsistent with Uniform.  So perhaps we should add an Exception to require xmin<=xmax (like we require a<=b for Uniform).

**Comment by hayate0304** [comment]
I just added it.

**Comment by dlsun** [comment]
Kien, thanks for these changes. It was too complicated to try to merge this PR after all the changes that had been made, so I just made the changes manually in a separate commit: https://github.com/dlsun/symbulate/commit/e6450d45635e935638018c0a282c9c6a4de4499d

The major changes are:
- Changing the default parametrization for the Normal from `var=` to `sd=`
- Making heat map color scales start from 0.
- Replacing `normed=True` by `density=normalize` to eliminate deprecation warnings.
- Have `normalize=False` return counts for histograms.

**Pull request closed without merging**